pkg/kubelet/cm/dra/plugin/plugin.go

```diff
@@ -179,3 +188,32 @@ func newMetricsInterceptor(pluginName string) grpc.UnaryClientInterceptor {
 		return err
 	}
 }
+
+func (p *Plugin) HandleConn(ctx context.Context, connStats stats.ConnStats) {
+	logger := klog.FromContext(ctx)
+	switch connStats.(type) {
+	case *stats.ConnBegin:
+		logger.V(2).Info("Connection begin", "plugin", p.name, "endpoint", p.endpoint, "stats", connStats)
+		if p.connectionDropped {
+			if p.registrationHandler != nil {
+				logger.V(2).Info("Stopping ResourceSlices cleanup", "plugin", p.name, "endpoint", p.endpoint)
+				p.registrationHandler.Stop()
+			}
+			p.connectionDropped = false
+		}
+	case *stats.ConnEnd:
+		logger.V(2).Info("Connection end", "plugin", p.name, "endpoint", p.endpoint, "stats", connStats)
+		p.connectionDropped = true
+		if p.registrationHandler != nil {
+			p.registrationHandler.cleanupResourceSlices(p.name)
```

Comment:
@pohly e2e tests are flaking probably because this call conflicts with the one called when plugin is un-registered. I'm looking at it, but your help would be appreciated as you know that code better.

Reply:
See https://github.com/kubernetes/kubernetes/pull/131073/files#r2104086671. I think state tracking is simply incomplete at the moment.